tasmota/xsns_88_tfminiplus.ino

```diff
@@ -0,0 +1,258 @@
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+// Conditional compilation of driver
+#ifdef USE_TFMINIPLUS
+
+/*****************************************************************************\
+ * TFmini, TFmini Plus, TFmini Plus (Indoor Version), TFmini-S - LiDAR Module
+ * Manufacturer: Benewake (Beijing) Co. Ltd.
+ *
+ * Code for Time-Of-Flight (TOF) family single-point short-range LiDAR sensor
+ * modules with UART interface.
+ * 
+ * Connection Description for GH1.25-4P (Molex51021-0400) connector:
+ *      No     Color       Corresponding PIN   Function    Comment
+ *      1       RED         PIN-1               +5V         Power Supply
+ *      2       WHITE       PIN-2               RxD/SDA     Receiving/Data
+ *      3       GREEN       PIN-3               TxD/SCL     Transmitting/Clock
+ *      4       BLACK       PIN-4               GND         Ground
+ * 
+ * Before connecting module to Tasmota, please prepare module with
+ * the following sequence through terminal program (e.g. Termite):
+ * 1. Connect sensor to USB-to-UART (TTL 3.3V)
+ * 2. Set terminal to 115200bps and 8N1
+ * 3. Set frame rate to 0 (zero):
+ *      send -->: 5A0603000000
+ *      receive <-- (OK): 5A0603000000
+ * 4. Set baud rate to 9600bps
+ *      send -->: 5A0806802500000D
+ *      receive <-- (OK): 5A0806802500000D
+ * 5. Set terminal to 9600bps and 8N1
+ * 6. Save settings
+ *      send -->: 5A04116F
+ *      receive <-- (OK): 5A05110070
+ *      receive <-- (FAIL): 5A05110071
+ * 7. When point 6 returned OK, than ready to connect to Tasmota! :)
+ *    When point 6 returned FAIL, start with item 1 and
+ *      - check connection of RX and TX pin are not exchanged
+ *      - check power supply 
+ *          - 5V +-0,5V
+ *          - 110mA average
+ *          - 140mA max peak
+ * 
+ * References:
+ * - http://en.benewake.com/product
+ * - https://de.aliexpress.com/item/32852024277.html?spm=a2g0s.9042311.0.0.27424c4d5Edizk
+ * - https://de.aliexpress.com/item/4001076614996.html?spm=a2g0s.9042311.0.0.27424c4d5Edizk
+\*****************************************************************************/
+
+#include <TasmotaSerial.h>
+
+// Define driver ID
+#define XSNS_88 88
+
+// Use special no wait serial driver, should be always on
+#ifndef ESP32
+#define SPECIAL_SS
+#endif
+
+// Default baudrate
+#define BAUDRATE 9600
+
+// Serial buffer length for incoming data
+#define TFMP_MAX_DATA_LEN 9
+
+char Tfmp_buffer[TFMP_MAX_DATA_LEN + 1];
+
+struct xsns_88_tfminiplus
+{
+    bool ready = false;
+    uint16_t distance = 0;
+    uint16_t sigstrength = 0;
+    uint16_t chiptemp = 0;
+} tfminiplus_sensor;
+
+// Software and hardware serial pointers
+#ifdef ESP8266
+TasmotaSerial *TfmpSerial = nullptr;
+#endif  // ESP8266
+#ifdef ESP32
+HardwareSerial *TfmpSerial = nullptr;
+#endif  // ESP32
```

Comment:
No need for this. TasmotaSerial does use hardware serial on ESP32 only.

Reply:
Ok, fine for me.